src/python/pants/core/goals/test.py

```diff
@@ -824,7 +807,9 @@ async def run_tests(
 
     environment_names = await MultiGet(
         Get(
-            EnvironmentName, {batch: TestRequest.Batch, local_environment_name.val: EnvironmentName}
```

Comment:
This callsite is the relevant one: the input `EnvironmentName` was immediately used, as if by running `QueryRule(EnvironmentName, [EnvironmentName])`.